core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -860,6 +860,14 @@ private[spark] class Executor(
     }
   }
 
+  private def getIsolatedSession(
+      taskDescription: TaskDescription) = {
+    taskDescription.artifacts.state match {
+      case Some(jobArtifactState) =>
+        isolatedSessionCache.get(jobArtifactState.uuid, () => newSessionState(jobArtifactState))
+      case _ => defaultSessionState
+    }
+  }
```

Comment:
Nit: This refactoring is unnecessary here. This also modifies [OSS-synced code](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/executor/Executor.scala#L516-L521) so we should leave it as it is